### PR TITLE
replace exprt::opX by named methods

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [X] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by chrisr-diffblue — APPROVED** [review]

File: src/goto-programs/builtin_functions.cpp [hunk 1]
```diff
@@ -1402,9 +1402,11 @@ void goto_convertt::do_function_call_symbol(
...
 
     // build (*ptr==oldval)?newval:*ptr
-    if_exprt if_expr(equal, arguments[2], deref_ptr, deref_ptr.type());
-    if(if_expr.op1().type()!=if_expr.type())
-      if_expr.op1().make_typecast(if_expr.type());
+    if_exprt if_expr(
+      equal,
+      typecast_exprt::conditional_cast(arguments[2], deref_ptr.type()),
+      deref_ptr,
+      deref_ptr.type());
```
Comment (tautschnig) [inline src/goto-programs/builtin_functions.cpp:1409]: If I could get a review of #3991 we wouldn't be changing the same code...

File: src/goto-programs/builtin_functions.cpp [hunk 2]
```diff
@@ -1461,9 +1463,11 @@ void goto_convertt::do_function_call_symbol(
...
```
Comment (tautschnig) [inline src/goto-programs/builtin_functions.cpp:1470]: #3991...

**Review by tautschnig — APPROVED** [review]
Just a request about the commit message: I think it needs to say "replace _uses of_ exprt::opX ...".

**Review by allredj — COMMENTED** [review]
:no_entry_sign:
This PR failed Diffblue compatibility checks (cbmc commit: 8fd215d89b64d6ddc8d5632096e921dd01d2c144).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/99956731
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Pull request merged**